annotation-fulltext/src/main/java/eu/europeana/annotation/fulltext/subtitles/SubtitleHandler.java

```diff
@@ -1,52 +1,64 @@
 
+import eu.europeana.annotation.config.SubtitleFormat;
+import eu.europeana.annotation.config.SubtitleFormats;
+
+@Component("subtitleHandler")
 public class SubtitleHandler {
 	
-	private static final String INTERNET_MEDIA_TYPE_VTT_FORMAT = "text/vtt";
-	private static final String INTERNET_MEDIA_TYPE_DFXP_FORMAT = "application/ttml+xml";
-	private static final String INTERNET_MEDIA_TYPE_QT_FORMAT = "video/quicktime";
-	
-    private static final SubtitleImportHandler QT_HANDLER = new QtTextImportHandler();
-    private static final SubtitleImportHandler DFXP_HANDLER = new DfxpImportHandler();
-    private static final SubtitleImportHandler VTT_HANDLER = new WebVttImportHandler();
+	public SubtitleHandler(String subtitleFormatsXMLConfigFile, XmlMapper xmlJacksonMapper) throws IOException {
+		try (InputStream inputStream = getClass().getResourceAsStream(subtitleFormatsXMLConfigFile);
+	    		BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
+	    	    String contents = reader.lines().collect(Collectors.joining(System.lineSeparator()));
+	    	    subtitlesFormats = xmlJacksonMapper.readValue(contents, SubtitleFormats.class);  
+    	}
+	}
+
+	SubtitleFormats subtitlesFormats;
 	
-	public List<SubtitleItem> parseSubtitle (String text, String format) throws FileFormatException, IOException {
+	public List<SubtitleItem> parseSubtitle (String text, String format) throws FileFormatException, IOException, ClassNotFoundException, NoSuchMethodException, SecurityException, InstantiationException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
 		if (StringUtils.isBlank(text)) {
 			return null;
-		}
-		List<SubtitleItem> items = new ArrayList<SubtitleItem>();
-		InputStream stream = new ByteArrayInputStream(text.getBytes(StandardCharsets.UTF_8));
-		switch(format) {
-		  case INTERNET_MEDIA_TYPE_QT_FORMAT:
-			  QT_HANDLER.importFile(stream, new Configuration());		    
-			  break;
-		  case INTERNET_MEDIA_TYPE_DFXP_FORMAT:
-		      DFXP_HANDLER.importFile(stream, new Configuration());
-		      break;
-		  case INTERNET_MEDIA_TYPE_VTT_FORMAT:
-		      VTT_HANDLER.importFile(stream, new Configuration());
-		      break;
-		  default:
-			  throw new FileFormatException("The Internet media subtitle format does not match any of the expected formats: "+ 
-				  INTERNET_MEDIA_TYPE_QT_FORMAT+","+INTERNET_MEDIA_TYPE_DFXP_FORMAT+","+INTERNET_MEDIA_TYPE_VTT_FORMAT+".");
 		}		
+				
+		Optional<SubtitleFormat> subtitleFormat = subtitlesFormats.getFormats().stream().filter(s -> format.equals(s.getMimetype())).findFirst();
+		if (subtitleFormat.isEmpty()) return null;
+		
+		Class<?> subtitleHandlerClass = Class.forName(subtitleFormat.get().getHandler());
+		Constructor<?> constructorSubtitleHandlerClass = subtitleHandlerClass.getConstructor();
+		SubtitleImportHandler subtitleImportHandler = (SubtitleImportHandler) constructorSubtitleHandlerClass.newInstance(new Object[] {});
```

Comment:
The handlers must be concerted to (singleton) beans for performance reasons

Reply:
Done.